core/model/model.go

```diff
@@ -114,3 +116,16 @@ func (u UUID) Validate() error {
 	}
 	return nil
 }
+
+// HostedModel bundles a Model with its cloud and credential.
+type HostedModel struct {
+	Model
```

Comment:
These should have field names rather than being embedded, it can lead to a lot of confusion.